tests/net/webrequest.cpp

```diff
@@ -0,0 +1,213 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        tests/net/webrequest.cpp
+// Purpose:     wxWebRequest test
+// Author:      Tobias Taschner
+// Created:     2018-10-24
+// Copyright:   (c) 2018 wxWidgets development team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+#include "testprec.h"
+
+#ifdef __BORLANDC__
+#pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/wx.h"
+#endif // WX_PRECOMP
+
+#include "wx/webrequest.h"
+#include "wx/filename.h"
+#include "wx/wfstream.h"
+
+// This test requires the URL to an httpbin instance.
+// httpbin is a HTTP Request & Response Service available at:
+// https://httpbin.org
+// It can also be run locally via a simple docker run
+//
+// For this test to run the the base URL has to be specified with
+// an environment variable, e.g.:
+//   WX_TEST_WEBREQUEST_URL=https://httpbin.org
```

Comment:
We need to update `build/tools/travis-ci.sh` and `build/tools/appveyor-test.bat` to do this.